### PR TITLE
Removed the connectedCheck

### LINKED ISSUE #1181: Circle CI Build Failing
**Actual Behaviour**

Circle CI build is failing most of the time
```
java.lang.RuntimeException: Waited for the root of the view hierarchy to have window focus and not request layout for 10 seconds. If you specified a non default root matcher, it may be picking a root that never takes focus. Root:
```

**Expected Behaviour**

Seems like the root view isn't getting focus as some other screen might be obstructing it. In short, the UI tests are failing and need to be fixed.

**Would you like to work on the issue?**
No. Comment if you want to work.

### PR DESCRIPTION
**Pull request by batbrain7**
Fixes #1181

### CODE REVIEW

**Comment by batbrain7** [comment]
@iamareebjamal please review

File: circle.yml [hunk 1]
```diff
@@ -19,11 +19,9 @@ test:
             parallel: true
         
         # Wait for it to have booted
-        - circle-android wait-for-boot
-        
+
         # Tun tests  against the emulator.
-        - ./gradlew connectedCheck
-        
+
```
Comment (iamareebjamal) [inline circle.yml]: Remove comments, add `./gradlew build`

Remove
```
emulator -avd circleci-android22 -no-audio -no-window:
            background: true
parallel: true
```

Comment (iamareebjamal) [reply circle.yml]: Remove

```
- mkdir $CIRCLE_TEST_REPORTS/AndroidTests
        - cp -r app/build/reports/androidTests/connected/* $CIRCLE_TEST_REPORTS/AndroidTests
```

And replace this with unit tests report

Comment (iamareebjamal) [reply circle.yml]: Change this to correct build apk directory `cp -r ${HOME}/${CIRCLE_PROJECT_REPONAME}/app/build/outputs/apk/ $CIRCLE_ARTIFACTS`

**Comment by iamareebjamal** [comment]
Also add caching, see Orga App circle config for details

File: circle.yml (hunk 1 above)
Comment (batbrain7) [reply circle.yml]: @iamareebjamal unit tests are not yet initialised so path are you referring to?

Comment (batbrain7) [reply circle.yml]: which path*

Comment (iamareebjamal) [reply circle.yml]: Unit tests are ran when build command is executed

Comment (batbrain7) [reply circle.yml]: so the reports path will ci path starting with usr/ubuntu/  ?

Comment (iamareebjamal) [reply circle.yml]: No, just the relative path will change. Please take a look at Orga App config

**Comment by batbrain7** [comment]
@chiragw15 the config file here is a little different than from the orga app's file
Please assist here

**Comment by iamareebjamal** [comment]
@batbrain7 Just look at the test report path. Also look at the circle CI config format. Orga App is using v2 of that. You can use the same

**Comment by batbrain7** [comment]
@iamareebjamal updating the config .yml file completely then

File: circle.yml [hunk 2]
```diff
@@ -1,40 +1,33 @@
...
-        - mkdir $CIRCLE_TEST_REPORTS/Lint
-        - cp app/build/reports/lint-results.* $CIRCLE_TEST_REPORTS/Lint
-
-        # Copy tests report
-        - mkdir $CIRCLE_TEST_REPORTS/AndroidTests
-        - cp -r app/build/reports/androidTests/connected/* $CIRCLE_TEST_REPORTS/AndroidTests
-
-        - bash exec/prep-key.sh
-        - chmod +x exec/apk-gen.sh
-        - ./exec/apk-gen.sh
```
Comment (iamareebjamal) [inline circle.yml]: This should be added to upload the apk

**Comment by batbrain7** [comment]
@iamareebjamal please review and merge

**Comment by iamareebjamal** [comment]
Strange that unit tests did not run on build. Try locally and check what's happening

**Comment by iamareebjamal** [comment]
The report for JUnit tests should be generated on build. If it is not being generated, then explicitly run `./gradlew test`

**Comment by batbrain7** [comment]
@iamareebjamal build successfully passed on ./gradlew test command, should i add this command ?

**Comment by iamareebjamal** [comment]
It should not just pass, there should be generated test report in the build folder as well

**Comment by batbrain7** [comment]
@iamareebjamal  in the app there is only an androidTest package, I found it weird not finding a test package

**Comment by iamareebjamal** [comment]
That means even unit tests are in androidTest package, please move them locally and try

**Comment by iamareebjamal** [comment]
It seems like there are no unit tests at all. Just fix the line and change the requested for now. Thanks

**Comment by batbrain7** [comment]
Yes @iamareebjamal there are no unit tests at all

File: circle.yml [hunk 3]
```diff
@@ -1,40 +1,33 @@
...
+          key: jars-{{ checksum "build.gradle" }}-{{ checksum  "app/build.gradle" }}
+      - run:
+          name: Download Dependencies
+          command: ./gradlew androidDependencies
+      - save_cache:
+          paths:
+            - ~/.gradle
+          key: jars-{{ checksum "build.gradle" }}-{{ checksum  "app/build.gradle" }}
+      - run:
+          name: Run Tests
```
Comment (iamareebjamal) [inline circle.yml]: Change to `lint check`

File: circle.yml [hunk 4]
```diff
@@ -1,40 +1,33 @@
...
+          command: ./gradlew lint
+      - run:
+          name: Test Coverage
```
Comment (iamareebjamal) [inline circle.yml]: Change to Run tests

File: circle.yml (hunk 2 above)
Comment (iamareebjamal) [reply circle.yml]: This is not resolved

File: circle.yml [hunk 5]
```diff
@@ -1,40 +1,36 @@
...
+          path: app/build/reports
+          destination: reports
+      - store_artifacts:
+          path: app/build/outputs
+          destination: outputs
+      - store_test_results:
+          path: app/build/test-results
+      - bash exec/prep-key.sh
+      - chmod +x exec/apk-gen.sh
+      - ./exec/apk-gen.sh
```
Comment (iamareebjamal) [inline circle.yml]: This is not how it works. You have to add it in run block in command section like given above for tests and lint

File: circle.yml (hunk 2 above)
Comment (batbrain7) [reply circle.yml]: in what format this be added? I tried adding these at the end but ci build showed 
this error : orb not found: bash exec

File: circle.yml (hunk 5 above)
Comment (batbrain7) [reply circle.yml]: Doing the changes

**Comment by batbrain7** [comment]
@iamareebjamal CI fail shows permission denied error could adding su or sudo fix it?

**Comment by batbrain7** [comment]
@iamareebjamal done the changes

**Comment by batbrain7** [comment]
@iamareebjamal please merge this, now we can proceed towards the Unit tests

**Pull request merged**